pandas/_typing.py

```diff
@@ -125,12 +125,12 @@
 ]
 
 # dtypes
-NpDtype = Union[str, np.dtype, type_t[Union[str, complex, bool, object]]]
+NpDtype = Union[str, "np.dtype[Any]", type_t[Union[str, complex, bool, object]]]
```

Comment:
Probably safer to change it to `np.dtype[np.generic]` as np.dtype is bound by np.generic

https://github.com/numpy/numpy/blob/2a6daf39cc4fd895ab803edf018907cb8044f821/numpy/__init__.pyi#L681

Reply:
unfortunately, mypy fails now because `np.generic` is not covered by `Scalar`. I would be fine with ignoring the error for now:
```py
# error: Incompatible types in assignment (expression has type "Union[generic, Any]", variable has type "Union[Union[str, float, bool], Union[Period, Timestamp, Timedelta, Interval[Any]], datetime64, timedelta64, datetime]")
bad_line  # type: ignore[assignment]
```

Reply:
Then I think for now we can just make it `Any`, and add a note to this? What do you think about this/

Reply:
using `Any` hides a legit inconsistency we have between `NpDtype` and `Scalar`, so I would prefer having this error - even if it is ignored for now.

Reply:
sg. Let me update this.

Reply:
you will need to add the ignore comments :) https://github.com/pandas-dev/pandas/pull/48375#discussion_r962862828